migrations/account_storage.go

```diff
@@ -68,23 +71,17 @@ func (i *AccountStorage) ForEachValue(
 
 			value := storageMap.ReadValue(nil, storageKey)
 
-			newValue, updated := valueConverter(value)
-			if newValue == nil && !updated {
+			newValue := valueConverter(value, i.address, domain, key)
+			if newValue == nil {
 				continue
 			}
 
-			if newValue != nil {
-				// If the converter returns a new value, then replace the existing value with the new one.
-				storageMap.SetValue(
-					inter,
-					storageKey,
-					newValue,
-				)
-			}
-
-			if reporter != nil {
-				reporter.Report(i.address, domain, key)
-			}
```

Comment:
Any reason for removing the reporting here?

Reply:
It is moved to `StorageMigration` at: https://github.com/onflow/cadence/blob/3382e8db54825f25d589342215384dda03c263ad/migrations/migration.go#L209-L211

so the migration does the reporting, rather than the storage iterator